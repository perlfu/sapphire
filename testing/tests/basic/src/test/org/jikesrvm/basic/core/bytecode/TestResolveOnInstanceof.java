package test.org.jikesrvm.basic.core.bytecode;

/**
 * According the the definition of instanceof the method must be resolved prior to any other checks.
 * See http://java.sun.com/docs/books/jvms/second_edition/html/Instructions2.doc6.html#instanceof.Description
 *
 * Also described in [ 1147107 ]  unresolved instanceof etc. on &lt;null&gt; not compliant.
 *
 * @author Peter Donald
 */
public class TestResolveOnInstanceof {

  static interface A {
    String genString();
  }

  public static void main(String[] args) {
    System.out.println(doInstanceOf(new Object()));
  }

  static boolean doInstanceOf(final Object a) {
    return a instanceof A;
  }
}
