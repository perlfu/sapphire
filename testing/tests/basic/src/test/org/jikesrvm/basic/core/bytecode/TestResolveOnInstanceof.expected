Registering transformer
Transforming class: test/org/jikesrvm/basic/core/bytecode/TestResolveOnInstanceof
Transforming class: test/org/jikesrvm/basic/core/bytecode/TestResolveOnInstanceof$A
false
